scripts/monitoring/cron-send-elb-health.py

```diff
@@ -0,0 +1,182 @@
+
+def elb_instance_health(instance_state, instance_name, elb_name):
+    ''' Check health of each instance '''
+    if instance_state != instance_healthy:
+        unhealthy_detected = {
+                "elb": elb_name,
+                "instance": instance_name,
+                "state": instance_state,
+                }
+        elb_instance_unhealthy_count.append(unhealthy_detected)
+
+def elb_health_check(client, elbs_discovered):
+    ''' Check health of each node found behind each ELB '''
+
+    # Iterate through raw health checks of each instance behind each ELB
+    for i in range(len(elbs_discovered)):
+        elb_health_checks_raw = client.describe_instance_health(
+                LoadBalancerName=elbs_discovered[i]
+                )
+
+        # Get https response
+        elb_response_http = elb_health_checks_raw['ResponseMetadata']['HTTPStatusCode']
+
+        # Get instance health/state
+        elb_instance_response_states = elb_health_checks_raw['InstanceStates']
+
+        # Check count of instances behind each ELB. Alert on 0 count.
+        elb_instance_count(elb_instance_response_states, elbs_discovered[i])
+
+        # Iterate through each instances health/state behind the ELB
+        for elb_instance_response_state in elb_instance_response_states:
+            elb_name = elbs_discovered[i]
+            elb_instance_name = elb_instance_response_state['InstanceId']
+            elb_instance_state = elb_instance_response_state['State']
+
+            # Check http response
+            if elb_response_http != 200:
+                print("A potential error occurred. HTTP Response: %s" % elb_response_http)
+
+            elb_instance_health(elb_instance_state, elb_instance_name, elb_name)
+#DEBUG            print("ELB: %s Instance: %s State: %s" % (elb_name, elb_instance_name, elb_instance_state))
+
+def main():
+    ''' Gather and examine details about this node within ELBs '''
+
+    args = parse_args()
+
+#    aws_access, aws_secret = get_aws_creds('~/.aws/credentials')
+    #instance_region = get_instance_region()
+    aws_access, aws_secret = get_aws_creds('./credentials')
+    instance_region = "eu-west-2"
+
+    # Create boto client to access ELB resources
+    client = boto3.client(
+            'elb',
+            aws_access_key_id=aws_access,
+            aws_secret_access_key=aws_secret,
+            region_name=instance_region
+            )
+
+    # Call all available loadbalancers and store blob result in elb_descriptions
+    elb_descriptions = client.describe_load_balancers()
+
+    # Get a list of available ELBs
+    elbs_discovered=discover_elbs(elb_descriptions)
+
+    # Perform health check of each instance available behind each ELB
+    elb_health_check(client, elbs_discovered)
+
+    ### Metric Checks
+    if len(elb_no_instances) != 0:
+        for elb in range(len(elb_no_instances)):
+            print("ELB: %s has no instances behind it. Please investigate." % elb_no_instances[elb])
```

Comment:
I think this should increment `elb_instance_unhealthy_count_metric` 

Having no instances behind an ELB should be a fault condition.